transaction/core/src/blockchain/block.rs

```diff
@@ -122,7 +123,7 @@ impl Block {
     /// * `root_element` - The root element for membership proofs
     /// * `block_contents` - Contents of the block.
     pub fn new(
-        version: u32,
+        version: BlockVersion,
```

Comment:
we could consider that `Block` doesn't use a `BlockVersion` and uses a `u32` instead, since it is using u32 internally. that would make the ledger db test around adding too recent blocks still possible.

Reply:
+1 for keeping BlockVersion and using the hack in test